crates/core_arch/src/aarch64/neon/mod.rs

```diff
@@ -276,12 +312,160 @@ pub unsafe fn vabsq_s64(a: int64x2_t) -> int64x2_t {
     vabsq_s64_(a)
 }
 
+/// Add pairwise
+#[inline]
+#[target_feature(enable = "neon")]
+#[cfg_attr(test, assert_instr(addp))]
+pub unsafe fn vpaddq_s16(a: int16x8_t, b: int16x8_t) -> int16x8_t {
+    vpaddq_s16_(a, b)
```

Comment:
This can use `simd_add`, right?

Reply:
No, that does a normal "vertical" add, this is a pairwise add.
```
// this is simd_add
vertical_add([a, b, c, d], [e, f, g, h]) => [a+e, b+f, c+g, d+h]

// this is the padd op family
pairwise_add([a, b, c, d], [e, f, g, h]) => [a+b, c+d, e+f, g+h]
```